parsers/lib/antlr4/WithPartsCommentedOut_LangPVLParser.g4.bak

```diff
@@ -0,0 +1,335 @@
+ | relExpr '>=' setExpr
+ | relExpr '>' setExpr
+ | setExpr
+ ;
+
+setExpr
+ : setExpr 'in' addExpr
+ | addExpr
+ ;
+
+addExpr
+ : addExpr '+' multExpr
+ | addExpr '-' multExpr
+ | multExpr
+ ;
+
+multExpr
+ : multExpr '*' powExpr
+ | multExpr '/' powExpr
+ | multExpr '%' powExpr
+ | multExpr '\\' powExpr
+ | powExpr
+ ;
+
+powExpr
+ : powExpr '^^' seqAddExpr
+ | seqAddExpr
+ ;
+
+seqAddExpr
+ : unaryExpr ('::' seqAddExpr)?
+ | seqAddExpr '++' unaryExpr
+ | seqAddExpr '++' '(' unaryExpr ',' unaryExpr ')'
+// | unaryExpr
+ ;
+
+unaryExpr
+ : '!' unaryExpr
+ | '-' unaryExpr
+ | newExpr
+ ;
+
+newExpr
+ : 'new' identifier tuple
+ | 'new' nonArrayType newDims
+ | nonTarget
+// | target
+ ;
+
+nonTarget
+ : nonTarget '.' gen_id
+ | nonTarget tuple
+ | nonTarget '[' '..' expr ']'
+ | nonTarget '[' expr ']'
+ | nonTarget '[' expr '..' expr? ']'
+ | nonTarget '[' expr '->' expr ']'
+ | nonTarget '->' identifier tuple
+ | nonTargetUnit
+ ;
+
+target
+ : target '.' gen_id
+ | target '[' expr ']'
+ | nonTarget '.' gen_id
+ | nonTarget '[' expr ']'
+ | targetUnit
+ ;
+
+nonTargetUnit
+ : valPrimary
+ | 'this'
+ | 'null'
+// | 'true'
+// | 'false'
+ | 'current_thread'
+// | '\\result'
+
+ | collectionConstructors // About 100 ambiguities
+ | 'map' '<' type ',' type '>' mapValues // -5 ambiguities?
+ | 'tuple' '<' type ',' type '>' values // 0 ambiguities
+// | builtinMethod tuple // 49 ambiguities
+ | '\\owner' '(' expr ',' expr ',' expr ')'
+ | 'id' '(' expr ')'
+
+// | '|' expr '|'
+// | '?' identifier // Duplicate in builtinMethod
+ | NUMBER
+ | values
+ | '(' expr ')'
+ | identifier
+ ;
+
+collectionConstructors
+ : container '<' type '>' values
+ | '[' exprList ']'
+ | '[t:' type ']'
+ | '{' exprList '}'
+ | '{t:' type '}'
+ | 'b{' exprList '}'
+ | 'b{t:' type '}'
+ | 'set' '<' type '>' '{' expr '|' setCompSelectors ';' expr '}'
+ ;
+
+targetUnit
+ : identifier
+ ;
+
+builtinMethod
+ : ('Value' | 'HPerm' | 'Perm' | 'PointsTo' | 'Hist' | '\\old' | '?' | 'idle' | 'running' | 'head' | 'tail' | 'held' | 'Some')
+ ;
+
+values : '{' exprList? '}';
+
+mapValues : '{' mapPairs? '}';
+
+tuple : '(' exprList? ')';
+
+block : '{' statement* '}' ;
+
+setCompSelectors
+    : type identifier
+    | type identifier '<-' identifier
+    | type identifier '<-' collectionConstructors
+    | type identifier ',' setCompSelectors
+    | type identifier '<-' identifier ',' setCompSelectors
+    | type identifier '<-' collectionConstructors ',' setCompSelectors
+    ;
+
+statement
+ : 'return' expr? ';'
+ | 'lock' expr ';'
+ | 'unlock' expr ';'
+ | 'wait' expr ';'
+ | 'notify' expr ';'
+ | 'fork' expr ';'
+ | 'join' expr ';'
+ | 'action' tuple block
+ | valStatement
+ | 'if' '(' expr ')' statement elseBlock?
+ | 'barrier' '(' identifier barrierTags? ')' barrierBody
+// | contract 'par' parUnitList
+ | 'vec' '(' iter ')' block
+ | 'invariant' identifier '(' expr ')' block
+ | 'atomic' '(' identifierList ')' block
+// | invariantList 'while' '(' expr ')' statement
```

Comment:
On different parts in this file there is commented code.

Reply:
This is the backup of the file I used to test/profile before I began to actually apply and integrate some performance tweaks I found. So it will be deleted entirely before this PR is merged.

Reply:
I didn't notice this was a draft before posting my comments.